dist/systemd/nomad.service

```diff
@@ -1,12 +1,16 @@
 [Unit]
 Description=Nomad
 Documentation=https://nomadproject.io/docs/
+Wants=network-online.target
+After=network-online.target
 
 [Service]
 KillMode=process
 KillSignal=SIGINT
 ExecStart=/usr/bin/nomad agent -config /etc/nomad
 ExecReload=/bin/kill -HUP $MAINPID
+Restart=on-failure
+RestartSec=30
```

Comment:
This seems awfully long. The Nomad agent must heartbeat script health checks otherwise they'll be marked as failing. Perhaps shorten this to a few seconds to prevent script checks from expiring?

Reply:
Okay. I will remove the `RestartSec` so that it defaults to 100ms.

Reply:
In my opinion 100 ms is way too quick. I personally use a RestartSec of 10 seconds

Reply:
I think default value of 100ms is Okay. May be it is on the aggressive side. But considering that this is the sleeping time before restarting the unit, the default value should be good enough.

Reply:
The reason I added the large delay was due to the scenario where sometimes the data directory gets corrupted after a machine boot.
The delay is also to let Consul startup.

Reply:
Hm perhaps a `RestartSec=2` would be sufficient for that and mentioning how to setup the Consul dependency in a comment like:

```ini
Wants=network-online.target
After=network-online.target

# If you're using Consul you should use the following Wants/After settings instead:
#Wants=consul.service
#After=consul.service
```

Reply:
Okay, Will update the PR with comments to have a dependency on Consul.

Should I also have 2 seconds of delay between restarts?

Reply:
Yeah, sounds good to me. Thanks for your patience! Any further tweaks we should open a new issue for discussion as what's in this PR is already a big improvement.